cli/pkg/kctrl/cmd/core/wait_flags.go

```diff
@@ -19,13 +19,14 @@ type WaitFlagsOpts struct {
 	AllowDisableWait bool
 	DefaultInterval  time.Duration
 	DefaultTimeout   time.Duration
+	WaitByDefault    bool
 }
 
 func (f *WaitFlags) Set(cmd *cobra.Command, flagsFactory FlagsFactory, opts *WaitFlagsOpts) {
-	if opts.AllowDisableWait {
-		cmd.Flags().BoolVar(&f.Enabled, "wait", true, "Wait for reconciliation to complete")
+	if opts.AllowDisableWait || !opts.WaitByDefault {
+		cmd.Flags().BoolVar(&f.Enabled, "wait", !opts.WaitByDefault, "Wait for reconciliation to complete")
 	}
-	f.Enabled = true
+	f.Enabled = !opts.WaitByDefault
```

Comment:
It should be `opts.WaitByDefault` instead of `!opts.WaitByDefault`, right?

Reply:
Ah yes, missed that